### ISSUE TITLE
Model Description

### ISSUE THREAD

**Issue by Ianni** [cardwing/Codes-for-Lane-Detection#90]
First, thank you for releasing your code and all the work you have put into it.

I have a few questions regarding your model. I am aware of the lanenet and SCNN papers and their respective code repositories, but when I look at your implementation I am not so sure in which way you "merged" both models. 

I thought inspecting your model using tensorboard will help me, but it actually confused me even more.

My first questions would be:

1.  Can you please describe how you merged Lanenet and SCNN. For example, I see that you adopted the hardcoded 4 lines (1 lane to the left and 1 to the right) from SCNN and maybe also why you chose to do so? 
2.  Can you please point me to the file where the pyramidal convolutions are implemented from SCNN? It seems like I am too blind to find it in your repo.

After adding some small tensorboard logging, I was confused that quantities like Accuracy get tracked in scopes called "test_0" and "tower_0". And even further, that they differ in both which contradicts my assumption that the tower_X is just there to facilitate multi gpu usage (I am running on only 1 GPU). 

3.  Can you please explain how your model interacts? Especially the test_0 and tower_0 scopes?
4.  Also, why is test_0 called test? It is definitely not exclusively used in validation. Or at least my tensorboard records values during training and validation in this scope.

Thank you in advance. I hope I did not miss any documentation.

TB Graph for reference
![png](https://user-images.githubusercontent.com/471066/55735758-4684b180-5a22-11e9-9a4a-63bedcbb0a6d.png)

**Comment by cardwing** [comment]
1. I just use the codebase of LaneNet (dataloader, network definition, etc) and implement SCNN-Tensorflow.
2. Please see [this file](https://github.com/cardwing/Codes-for-Lane-Detection/blob/master/SCNN-Tensorflow/lane-detection-model/encoder_decoder_model/vgg_encoder.py)
As to Q3 and Q4, please read the codes and figure them out by yourself.
